### PR TITLE
Change basicmodule to use Base POM module as a parent module

### PR DESCRIPTION
**Pull request by akolodziejski**
Parent POM is : 
https://github.com/openmrs/openmrs-contrib-maven-parent-module/blob/master/pom.xml

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -24,44 +24,28 @@
...
 			<artifactId>openmrs-web</artifactId>
-			<type>test-jar</type>
+            <scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.openmrs.test</groupId>
 			<artifactId>openmrs-test</artifactId>
 			<type>pom</type>
+            <scope>provided</scope>
 		</dependency>
```
Comment (rkorytkowski) [inline api/pom.xml]: Is it needed to include these dependencies? Did you try without them?

File: omod/pom.xml [hunk 2]
```diff
@@ -22,27 +22,28 @@
...
 			<groupId>org.openmrs.api</groupId>
 			<artifactId>openmrs-api</artifactId>
-			<type>jar</type>
+            <scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.openmrs.api</groupId>
 			<artifactId>openmrs-api</artifactId>
-			<type>test-jar</type>
+            <scope>provided</scope>
```
Comment (rkorytkowski) [inline omod/pom.xml]: These api dependencies should not be needed. They are defined in the parent pom.

File: omod/pom.xml [hunk 3]
```diff
@@ -22,27 +22,28 @@
...
 			<groupId>org.openmrs.web</groupId>
 			<artifactId>openmrs-web</artifactId>
-			<type>test-jar</type>
+            <scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.openmrs.test</groupId>
 			<artifactId>openmrs-test</artifactId>
 			<type>pom</type>
+            <scope>provided</scope>
```
Comment (rkorytkowski) [inline omod/pom.xml]: openmrs-test should not be needed as well.

File: pom.xml [hunk 4]
```diff
@@ -22,100 +28,25 @@
 		<module>omod</module>
 	</modules>
 
+    <properties>
+        <openmrs.version>1.12.0-SNAPSHOT</openmrs.version>
```
Comment (rkorytkowski) [inline pom.xml]: it's called openmrs.core.version in the parent.

File: pom.xml [hunk 5]
```diff
@@ -22,100 +28,25 @@
...
-				<groupId>org.openmrs.test</groupId>
-				<artifactId>openmrs-test</artifactId>
-				<version>${openMRSVersion}</version>
-				<type>pom</type>
-				<scope>test</scope>
+				<scope>provided</scope>
 			</dependency>
-		</dependencies>
+        </dependencies>
 	</dependencyManagement>
```
Comment (rkorytkowski) [inline pom.xml]: This whole section should go away. All of this is defined in the parent pom.

File: omod/pom.xml [hunk 6]
```diff
@@ -22,27 +22,28 @@
...
 			<groupId>org.openmrs.web</groupId>
 			<artifactId>openmrs-web</artifactId>
-			<type>jar</type>
+            <scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>org.openmrs.web</groupId>
 			<artifactId>openmrs-web</artifactId>
-			<type>test-jar</type>
+            <scope>provided</scope>
```
Comment (rkorytkowski) [inline omod/pom.xml]: You miss <classifier>tests</classifier> here.

File: pom.xml (hunk 4 above)
Comment (rkorytkowski) [inline pom.xml]: Also make it the latest release 1.11.0.

File: api/pom.xml (hunk 1 above)
Comment (akolodziejski) [inline api/pom.xml]: org.openmrs.web is missing in parent dependencies. can be found only in dependencyManagment block

File: pom.xml [hunk 7]
```diff
@@ -22,100 +28,7 @@
...
-			<name>OpenMRS Nexus Repository</name>
-			<url>http://mavenrepo.openmrs.org/nexus/content/repositories/public</url>
-			<snapshots>
-				<enabled>false</enabled>
-			</snapshots>
-		</pluginRepository>
-	</pluginRepositories>
-
+    <properties>
+        <openmrs.version>1.11.0</openmrs.version>
```
Comment (rkorytkowski) [inline pom.xml]: Powinno być \<openmrs.core.version\>1.11.0\<openmrs.core.version\>

File: api/pom.xml [hunk 8]
```diff
@@ -20,48 +20,11 @@
...
-		</dependency>
-		<dependency>
-			<groupId>org.openmrs.web</groupId>
-			<artifactId>openmrs-web</artifactId>
-			<type>jar</type>
-		</dependency>
+
 		<dependency>
 			<groupId>org.openmrs.web</groupId>
 			<artifactId>openmrs-web</artifactId>
```
Comment (rkorytkowski) [inline api/pom.xml]: Web dependency needs to go away from api.

**Comment by rkorytkowski** [comment]
I have just noticed that you moved all resources from api to omod. It's not correct.

**Comment by rkorytkowski** [comment]
BasicModuleActivator should also stay in api.

**Comment by rkorytkowski** [comment]
Merged with follow up fixes in https://github.com/openmrs/openmrs-module-basicmodule/commit/c4908afe9c0eedd60464d0b316509f1777ebdaa5

Thanks @akolodziejski!

**Pull request merged**